src/OpenTelemetry/Trace/BaseExportActivityProcessor.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="BaseExportActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Implements processor that exports <see cref="Activity"/> objects.
+    /// </summary>
+    public abstract class BaseExportActivityProcessor : ActivityProcessor
+    {
+        protected readonly ActivityExporter exporter;
```

Comment:
FxCop would probably generate a warning here about field being exposed. Should be...

`private readonly ActivityExporter exporter`
or
`protected ActivityExporter Exporter { get; }`

Reply:
I need to spend time learning this - will leave this as-is for now:
1. private field won't work as we expect derived class to access the exporter.
2. protected property seems to have lower perf and diagnosability (e.g. debug a dump)?

Reply:
@reyang Here's the doc on the warning I'm referring to: [CA1051](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1051?view=vs-2019). 

It has this note about the perf:
> Properties that just return the value of a private or internal field are optimized to perform on par with accessing a field; the performance gain from using externally visible fields instead of properties is minimal. Externally visible refers to public, protected, and protected internal (Public, Protected, and Protected Friend in Visual Basic) accessibility levels.